### ISSUE TITLE
Confusion Matrix

### ISSUE THREAD

**Issue by Auth0rM0rgan** [dgurkaynak/marvel-finetuning#5]
Hi,

First of all, thanks for the code. your code helps me a lot and I hope inception(V2, V3) will be ready very soon :)   
Secondly, I have 2 questions if you can help me to find out the solution.

1. How can I get the confusion matrix of your model to understand the performance of a classification model?
2. How can I get the separate accuracy of each categories?

My main problem is when I want to get confusion matrix, I need the `true_class` and `predicted_class` and I don't know how

There is a function for confusion matrix which is: 

> tf.confusion_matrix(labels, predictions,num_classess)

How am I going to feed this function?

Thanks in advance.

**Comment by dgurkaynak** [comment]
Hey @arminXerror,

Currently I'm not planning to work on Inception model any time soon, but any kind of PRs are welcome.

https://github.com/dgurkaynak/tensorflow-cnn-finetune/blob/125e343be28ec095c6885989d5d9a2b0b2474df5/vggnet/finetune.py#L67
Here, `tf.argmax(model.score, 1)` tensor is the predicted class, and `tf.argmax(y, 1)` is the true class. You can use these to feed `tf.confusion_matrix`. But this will calculate the confusion matrix for individual batch. You need to accumulate them in some way to get overall confusion matrix.

**Comment by Auth0rM0rgan** [comment]
Hey @dgurkaynak,

This is exactly what I have done and got an error.
I added these 2 lines after calculating accuracy:

```
correct_pred = tf.equal(tf.argmax(model.score, 1), tf.argmax(y, 1))
accuracy = tf.reduce_mean(tf.cast(correct_pred, tf.float32))

true_class = tf.argmax(y, 1)
predicted_class = tf.argmax(model.score, 1)
```
and I have added ` tf.confusion_matrix` in the session, right after printing Validation accuracy (line 140)

```
print("{} Validation Accuracy = {:.4f}".format(datetime.datetime.now(), test_acc))
print(sess.run(tf.confusion_matrix(true_class, predicted_class, num_classes)))
```

But I got an error : 

> InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'Placeholder_1' with dtype float and shape [128,3]


Can you help me to find the way to build Confusion Matrix? 

Thanks in advance.

**Comment by arminXerror** [comment]
Hey @dgurkaynak,

I find the way to calculate confusion matrix and fixed my error which, I mentioned in the previous post. But, I have confusion matrix in every epoch not the confusion matrix of the whole network, I think I have to find the way to accumulate them can you help me on this?

**Comment by Auth0rM0rgan** [comment]
Find the way to build the confusion matrix and accumulate them. Thanks
